src/main/scala/com/fulcrumgenomics/umi/CorrectUmis.scala

```diff
@@ -252,12 +262,26 @@ class CorrectUmis
 
   /** Given a UMI sequence and a set of fixed UMIs, report the best match. */
   private[umi] def findBestMatch(bases: String, umis: Array[String]): UmiMatch = {
-    val mismatchLimit = this.maxMismatches + minDistance
-    val mismatches    = umis.map(umi => Sequences.countMismatches(bases, umi))
-    val min           = mismatches.min
-    val matched       = (min <= maxMismatches) && (mismatches.count(m => m < min + this.minDistance) == 1)
-
-    UmiMatch(matched, umis(mismatches.indexOf(min)), min)
+    if (cacheSize > 0 && cache.containsKey(bases)) cache.get(bases)
```

Comment:
You're doing double work here by using `containsKey` and then `get`.  I'd rewrite to do something like 

```scala
val previous = if (cacheSize > 0) cache.get(bases) else None
previous match {
...
}
```